### ISSUE TITLE
calcite-tab-title: add a `label` attribute/prop for accessibility when only using an icon for the tab title

### ISSUE THREAD

**Issue by EmilyWells18** [Esri/calcite-components#13477]
**monday.com sync:** #10775297772

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Tab Title

### Summary

Currently, calcite-tab-title can include both text and icons or just text and remain accessible because the component uses visible text as its accessibility name. However, if a user creates a calcite-tab-title with only an icon, the accessibility name is lost.

For example:
`<calcite-tab-title
 icon-start="information"/>`

instead of 

`<calcite-tab-title
icon-start="information">
 information
 </calcite-tab-title>`

It would be valuable to support calcite-tab-title components that can be composed as:

text and icon
text only
icon only

while maintaining proper accessibility.

### Actual Behavior

`<calcite-tab-title
 icon-start="information"/>` provides no accessibility as there is no prop to provide the accessible name.

### Expected Behavior

`<calcite-tab-title` provides a prop for the accessible name which will be taken when the visible text of the element doesn't exist. e.g. in this example:

`<calcite-tab-title
 icon-start="information" accessibilityLabel="Information"/>` 

### Reproduction Sample

https://codepen.io/ewells18/pen/gbrJdyq?editors=100

### Reproduction Steps

There are 3 tabs in the CodePen example:

- One with text only
- One with text and an icon
- One with an icon only

1. Inspect each calcite-tab-title in DevTools.
2. Open the Accessibility pane.
3. Observe:
- The component relies on visible text for its accessibility name.
- There is no aria-label, aria-labelledby, or title attribute.
- For the tab with only an icon, there is no visible text, so the accessibility name is missing entirely.

### Reproduction Version

3.3.3

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by geospatialem** [comment]
Added the `design` label for consideration of if Calcite should allow for use cases of `tab-title` with only icon. _Believe_ this has come up in the past, but can't recall if this is a pattern we would support. However, it does appear that adding an `aria-label` to the `tab-title` provides context to AT (_although unsupported by Calcite_): https://codepen.io/geospatialem/pen/jEqodxy

If we should support the above, we should consider adding a `label` property/attribute to `tab-title` (`estimate - 3` in development).

cc @Esri/calcite-designers

**Comment by DintaMel** [comment]
@ashetland can you provide additional context?

**Comment by ashetland** [comment]
We agree that this isn't something we need to explicitly document as supported, but adding an `aria-label` to `tab-title` would be beneficial for AT in scenarios where users are already doing this.

**Comment by geospatialem** [comment]
> We agree that this isn't something we need to explicitly document as supported, but adding an `aria-label` to `tab-title` would be beneficial for AT in scenarios where users are already doing this.

Thanks @ashetland 🚀 - updated the issue title to reflect adding the `label` attribute/prop to the component.
